data/bookkeeping/block.go

```diff
@@ -467,6 +489,61 @@ func ProcessUpgradeParams(prev BlockHeader) (uv UpgradeVote, us UpgradeState, er
 	return upgradeVote, upgradeState, err
 }
 
+type bonusPlan struct {
+	baseRound     basics.Round
+	baseAmount    basics.MicroAlgos
+	decayInterval basics.Round
+}
+
+var bonusPlans = []bonusPlan{
+	0: {},
+	1: {
+		baseRound:  0, // goes into effect with upgrade
+		baseAmount: basics.Algos(2),
+		// 2.9 sec rounds gives about 10.8M rounds per year.
```

Comment:
Round times are not stable (and averages likely continue to decrease some).

Reply:
Right, I'm trying to anticipate the need to change these things as round times change. When a new "bonus plan" is installed, it should change the decay rate only, if the amount is unset.  I'm not sure whether we need to support changing the current bonus amount explicitly at upgrade time. Probably.

Reply:
Added a bunch of tests `TestBonusUpgrades`